src/main/java/uk/gov/hmcts/reform/sscs/service/CaseFlagMappingService.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.helper.FlagCode;
+import uk.gov.hmcts.reform.sscs.model.DescendantCaseData;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+
+import java.util.Map;
+import javax.validation.constraints.NotNull;
+
+@Service
+@RequiredArgsConstructor
+@SuppressWarnings("PMD.LawOfDemeter")
+public class CaseFlagMappingService {
+
+    private final Map<String, String> caseFlags;
+
+    public DescendantCaseData getCaseFlagMapping(@NotNull HearingWrapper hearingWrapper) {
+        SscsCaseData caseData = hearingWrapper.getOriginalCaseData();
+        return DescendantCaseData
+            .DescendantCaseDataBuilder()
+            .caseFlags(mapCaseFlags(caseData))
+            .build();
+    }
+
+    public Map<String, String> mapCaseFlags(SscsCaseData caseData) {
```

Comment:
There's still no need for a map. The enum has all the information you need by itself

Reply:
Echoing Del - we could just add the relevant enums to a list on the object and the calling code can pull the code from the enum directly.